src/task2/routes/mwares.ts

```diff
@@ -0,0 +1,42 @@
+import { RequestHandler, ErrorRequestHandler } from 'express';
+import { v4 as uuid } from 'uuid';
+
+import { userStorage } from '../user-storage';
+import { MESSAGES } from '../constants';
+
+export const checkIfUserExists: RequestHandler = (req, res, next) => {
+  if (!userStorage.isUserExists(req.params.id)) {
+    return next({
+      custom: true,
+      type: 'body',
+      error: MESSAGES.ERR_USER_DOES_NOT_EXIST,
+    });
+  }
+  next();
+};
+
+export const userDataPreparation: RequestHandler = (req, res, next) => {
+  const id = uuid();
+
+  if (userStorage.isUserExists(id) || !userStorage.isLoginFree(req.body.login)) {
+    return next({
+      custom: true,
+      type: 'body',
+      error: MESSAGES.ERR_USER_EXISTS,
+    });
+  }
+  req.body.id = id;
```

Comment:
I would separate validation stuff from user data generation

maybe it's better to keep id creation where you actually form the user object and save it